Source/multi.cpp

```diff
@@ -481,12 +481,19 @@ bool InitMulti(GameData *gameData)
 	return true;
 }
 
+uint64_t GenerateGameSeed()
+{
+	auto now = std::chrono::high_resolution_clock::now();
+	auto duration = now.time_since_epoch();
+	return static_cast<uint64_t>(duration.count());
+}

```

Comment:
I just get the feeling that using the highest resolution clock available on any given platform increases the chance of uniqueness among game seeds generated within the same timeframe between systems running on that platform. Furthermore, differing clock implementations among different platforms should only further increase that chance of uniqueness between systems running different platforms. I don't really see any benefit of deliberately using a potentially less precise clock. At best, it can only reduce the uniqueness characteristics between game seeds on the same platform.

Reply:
Do we expect system_clock to not necessarily have second precision?

Reply:
Actually, I take it back. ephphatha pointed out that MSVC uses steady_clock for high_resolution_clock, and C++ specifies that "it can be time since last reboot". That seems like a bad time value to use as a seed, actually.

Reply:
> Do we expect system_clock to not necessarily have second precision?

Looks like gcc and clang use nanosecond precision, msvc uses 100 nanosecond precision, no idea what other platforms use. The default resolution is implementation defined so the value from count() could be any of nanoseconds, milliseconds, seconds without converting to a specific duration. It's not really a problem, just inconsistent behaviour between platforms.

Could avoid the issue by seeding the global rng once and using it as a source for game seeds (e.g. copy the 128bit state from the global rng to sgGameInitInfo then use jump() to advance the global rng)

Reply:
> Actually, I take it back. ephphatha pointed out that MSVC uses steady_clock for high_resolution_clock, and C++ specifies that "it can be time since last reboot". That seems like a bad time value to use as a seed, actually.

yeah, better to have on based on real time or we could have devices that boot in to the game and often get the same value i guess